1-Draft/RFCnnnn-PowerShell-Configuration.md

```diff
@@ -0,0 +1,185 @@
+---
+RFC:  RFCnnnn
+Author:  Ilya Sazonov
+Status:  Draft
+SupercededBy:  N/A
+Version: 0.1
+Area: Engine
+Comments Due: 2/25/2018
+---
+
+# `PowerShell Core` configuration
+
+## Motivation
+
+Consumers, developers, and enterprise system administrators should be able to flexibly and conveniently configure PowerShell Core and applications based on it.
+
+A public PowerShell Core configuration API would allow a unified way to manage PowerShell-based applications in a consistent manner.
+
+## Specification
+
+`PowerShell Core` is configured using the following schemes:
+
+- On Windows - Group Policy Objects (GPO), Group Policy Preferences (GPP) and settings files.
+- On Unix - settings files.
+
+The settings files have `JSON` format.
+
+**Warning** The settings files differ from `PowerShell Core` _profile_ files, which are PowerShell scripts run at startup.
+
+Configuration schemes allow to customize `PowerShell Core` in the most flexible way:
+
+- Enterprise system administrators can use GPO, GPP and computer-wide settings files to apply approved configuration settings and mandatory security settings in a centralized manner. The same settings can be applied at user, application or startup levels.
+- Developers and consumers can use user, application and startup level settings files.
+
+### Settings locations
+
+`PowerShell Core` settings are grouped into `Policy settings` and `Regular settings`. Regular settings are normal configuration settings. Regular settings can be treated as default values. Policy settings is high priority and overlap regular settings. Policy settings are used by administrators to centrally manage applications.
+
+Location | Policy settings | Regular settings
+- | - | -
+File section | "PowerShell": { "PolicySettings": {...} } | "PowerShell": { "RegularSettings": {...} }
+File section | "OtherPowerShellApplication": { "PolicySettings": {...} } | "OtherPowerShellApplication": { "RegularSettings": {...} }
+Registry key | Software\Policies\PowerShellCore | Software\PowerShellCore
+Registry key | Software\Policies\PowerShellCore\OtherPowerShellApplication | Software\PowerShellCore\OtherPowerShellApplication
+
+### Priority of applying settings
+
+Because a configuration setting can be in several schemes, the setting wins according to the priority of its scheme.
+
+#### Priorities for Policy settings in descending order
+
+Scheme | Windows | Unix
+-| - | -
+GPO -> Computer Policy | HKLM\Software\Policies\PowerShellCore | /etc/powershell.config.json
+GPO -> User Policy | HKCU\Software\Policies\PowerShellCore | See `Comment A` below
+File -> Computer-Wide | %ProgramFiles%/PowerShell/powershell.config.json | /opt/Microsoft/powershell/powershell.config.json
```

Comment:
Is there a reason why computer-wide is under the PowerShell installation in `/opt` rather than under `/etc`. I believe @JamesWTruher was talking about this at some point as well